scripts/check-gen.sh

```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
```

Comment:
This script name is confusing. Its not really checking if we did run generate. It checks if there is any changes in the repo. (when running this after make generate - then the context is correct). I would rename this script and pass a parameter that will be the message. this way we can re-use if for other checks. not just the `generate`. but other.